Sources/Sentry/Public/SentryDefines.h

```diff
@@ -18,6 +18,123 @@
 #    define SENTRY_HAS_METRIC_KIT 0
 #endif
 
+#if __has_include(<UIKit/UIKit.h>)
```

Comment:
h: `__has_include` wont work here. Even if the project is not liked with UIKit. this returns true.

Reply:
I think it'll be cleaner to use SENTRY_HAS_UIKIT here, although they're equivalent. What it does is only defines the macros below for iOS/tvOS and Catalyst (confirmed by putting a `#warning blah` right under it and compiling for each platform)

Reply:
We still need UIKit.h to be included for all the parameter/value types from UIKit to still work. Including the header alone doesn't cause UIKit to be linked, so it's ok to do so. It's only the explicit message sends to UIKit classes that causes the linker to include dyld load commands. But it is still possible to include the header even if it won't be linked. The header delivers declarations, the load command loads the binary implementation.